public/js/actions/tests/newSource.js

```diff
@@ -1,14 +1,21 @@
 "use strict";
 
-const { actions, selectors, createStore } = require("../../util/test-head");
+const { constants, selectors, createStore } = require("../../util/test-head");
 const fixtures = require("../../test/fixtures/foobar.json");
-const { newSource } = actions;
 const { getSourceByActor } = selectors;
 const sourcesFixtures = fixtures.sources.sources;
 
 const store = createStore();
 const expect = require("expect.js");
 
+// Write our own `newSource` to bypass the batching logic.
+function newSource(source) {
```

Comment:
Yeah, wherever we throttle it'll make it harder to test. But if we move to just testing the "dumb" component, and `connect` implemented throttling then we would bypass it easily. I'm not exactly sure of this yet. Dispatching "raw" actions isn't that bad. This test is mainly testing the reducer, so it's not a big deal.

Not sure what the right solution is, so we'll see.